src/applications/disability-benefits/view-payments/components/view-payments-lists/payments/Payments.jsx

```diff
@@ -26,22 +27,28 @@ const Payments = ({
   textContent,
   alertMessage,
 }) => {
+  const location = useLocation();
+  const navigate = useNavigate();
   const [currentData, setCurrentData] = useState([]);
-  const [currentPage, setCurrentPage] = useState(1);
-  // Using `useRef` here to avoid triggering a rerender whenever these are
-  // updated
-  const totalPages = useRef(0);
-  const paginatedData = useRef([]);
+  const currentPage = new URLSearchParams(location.search).get('page') || 1;
+  const [totalPages, setTotalPages] = useState(0);
```

Comment:
I would use `useRef` here unless you want the page to re-render when you call `setTotalPages`